bundle/Resources/public/js/views/cof-contentcreationview.js

```diff
@@ -435,23 +439,43 @@ YUI.add('cof-contentcreationview', function (Y) {
                 itemTemplate = this.get('suggestedItemTemplate'),
                 documentFragment = Y.one(document.createDocumentFragment()),
                 pathSeparator = '/',
+                endLongPath = '...',
+                insidePathSeparator = pathSeparator + endLongPath + pathSeparator,
+                maxPathLength = 40,
                 renderedItem;
 
             locations.forEach(function (location) {
                 var locationPath = location.get('path'),
                     locationName = location.get('contentInfo').get('name'),
-                    longPath = pathSeparator;
+                    shortPath = pathSeparator,
+                    longPath = pathSeparator,
+                    substringEnd;
 
                 if (locationPath.length) {
                     longPath = locationPath.reduce(function (path, location) {
                         return path + location.get('contentInfo').get('name') + pathSeparator;
                     }, '');
                 }
 
+                if (locationPath.length >= 3) {
```

Comment:
Int 3 is used in several places, if this is something that might change, you could move this is to variable.